src/app/home-page/recent-events/recent-events.component.html

```diff
@@ -1,103 +1,189 @@
-      <div *ngFor="let event of events$ | async; last as last">
-        <div [ngSwitch]="event?.type">
-          <div *ngSwitchCase="EventType.ADDVERSIONTOENTRY">
-            <strong>{{ event.initiatorUser?.username }} </strong> created the {{ event.version?.referenceType | lowercase }}
-            <strong>{{ event.version?.name }}</strong> in {{ event?.tool || event?.apptool ? 'tool' : 'workflow' }}
-            <a
-              [routerLink]="
-                event?.apptool
-                  ? '/tools/' + event?.apptool?.full_workflow_path
-                  : event?.tool
-                  ? '/tools/' + event?.tool?.tool_path
-                  : '/workflows/' + event?.workflow?.full_workflow_path
-              "
-              >{{
-                (event?.tool | entryToDisplayName) || (event?.workflow | entryToDisplayName) || (event?.apptool | entryToDisplayName)
-              }}</a
+  <div *ngIf="events?.length > 0 || isLoading; else noEvents">
+    <div *ngFor="let event of events" class="size-small widget-list-items event" fxLayout fxLayoutAlign="space-between center">
+      <div fxLayout>
+        <a [routerLink]="'/users/' + event.initiatorUser?.username">
+          <img src="{{ event?.initiatorUser?.avatarUrl }}" alt="User avatar" />
+        </a>
+        <div [ngSwitch]="event.type" class="my-3">
+          <!-- Entries -->
+          <div *ngSwitchCase="EventType.PUBLISHENTRY" class="truncate-text-2">
+            <strong
+              ><a class="no-underline" [routerLink]="'/users/' + event.initiatorUser?.username">{{
+                event.initiatorUser?.username
+              }}</a></strong
+            >
+            published the {{ event | recentEvents: 'entryType' }}
+            <a [routerLink]="event | recentEvents: 'entryLink'">{{ event | recentEvents: 'displayName' }}</a>
+          </div>
+          <div *ngSwitchCase="EventType.UNPUBLISHENTRY" class="truncate-text-2">
+            <!-- Do not link to an entry that doesn't exist - 404 -->
+            <strong
+              ><a class="no-underline" [routerLink]="'/users/' + event.initiatorUser?.username">{{
+                event.initiatorUser?.username
+              }}</a></strong
             >
-            <div class="subtitle">{{ 'on ' + (event.dbCreateDate | date: 'medium') }}</div>
-            <mat-divider *ngIf="!last" class="custom-margin"></mat-divider>
+            unpublished the {{ event | recentEvents: 'entryType' }} <strong>{{ event | recentEvents: 'displayName' }}</strong>
           </div>
-          <div *ngSwitchCase="EventType.CREATECOLLECTION">
-            <strong>{{ event.initiatorUser?.username }}</strong> created the collection
-            <a [routerLink]="'/organizations/' + event.organization?.name + '/collections/' + event.collection?.name"
-              >{{ event.collection?.name }}
-            </a>
+          <div *ngSwitchCase="EventType.ADDVERSIONTOENTRY" class="truncate-text-2">
```

Comment:
This is better than before, but I wonder if we could further simplify by either:
1. creating templates for the various "categories" of events, and for each event type, pass the appropriate user/verb/phrase/etc to them.
2. creating some pipes to construct linked text to collections/orgs/events, eliminating much of the duplicate code.

Reply:
Was thinking about this, I think option 1 would be the most viable, would probably require creating a component for entry events, org events and collection events, and we would need to pass the `event` to the template as well to generate the links.  Some cases have additional information that we'd need to also have `*ngIf`s for.

Going to hold off on it because it would be a large refactor I'd prefer to do after this is merged, but will make a ticket for it.